pkg/api/handlers/compat/images_build.go

```diff
@@ -564,6 +569,12 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 			utils.BadRequest(w, "platform", platformSpec, err)
 			return
 		}
+		if os == "" {
+			os = runtime.GOOS
+		}
+		if arch == "" {
+			arch = runtime.GOARCH
+		}
```

Comment:
This overrides values computed from (possibly multiple) `Platform` values with the single values from `arch` and `os`, which looks wrong.  They should only be used in place of the values returned by `github.com/containerd/containerd/platforms.DefaultSpec()` if no `Platform` values are present, otherwise they should be ignored.

Reply:
I don't recall, why I added those, they do not look correct from reading parse.  I will remove and see if errors return.